### PR TITLE
adding models

### PR DESCRIPTION
**Pull request by marinaRomanchuk**

### CODE REVIEW

File: main/models.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from django.db import models
+
+
+class User(models.Model):
+    first_name = models.CharField(max_length=30)
+    last_name = models.CharField(max_length=50)
+    nickname = models.CharField(max_length=30, unique=True)
```
Comment (revimi) [inline main/models.py]: It is usually called `username`
```suggestion
    username = models.CharField(max_length=30, unique=True)
```

File: main/models.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    personal_information = models.TextField(max_length=300)
```
Comment (revimi) [inline main/models.py]:
```suggestion
    description = models.TextField(max_length=300)
```

File: main/models.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+from django.db import models
+
+
+class User(models.Model):
```
Comment (revimi) [inline main/models.py]: Please use [Django user model](https://docs.djangoproject.com/en/4.0/ref/contrib/auth/#user-model) or create a custom user model as described [here](https://docs.djangoproject.com/en/4.0/topics/auth/customizing/#specifying-a-custom-user-model). Please add `timestamp`, `date_of_birth`, `email` fields.

File: main/models.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        verbose_name_plural = 'Users'
+
+    def __str__(self):
+        return f'{self.nickname}'
+
+
+class Post(models.Model):
+    photo = models.ImageField()
+    description = models.TextField(max_length=500, blank=True)
+    published = models.DateTimeField(auto_now_add=True, db_index=True)
```
Comment (revimi) [inline main/models.py]:
```suggestion
    timestamp = models.DateTimeField(auto_now_add=True, db_index=True)
```

File: main/models.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    class Meta:
+        verbose_name = 'Comment'
+        verbose_name_plural = 'Comments'
+
+    def __str__(self):
+        return f'{self.comment_text}'
+
+
+class Like(models.Model):
```
Comment (revimi) [inline main/models.py:46]: Please add `timestamp` field

File: main/models.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    post = models.ForeignKey('Post', on_delete=models.CASCADE)
+    user = models.ForeignKey('User', on_delete=models.CASCADE)
+
+    class Meta:
+        verbose_name = 'Like'
+        verbose_name_plural = 'Likes'
+
+
+class Dislike(models.Model):
```
Comment (revimi) [inline main/models.py:56]: Please add `timestamp` field

File: main/models.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    post = models.ForeignKey('Post', on_delete=models.CASCADE)
+    user = models.ForeignKey('User', on_delete=models.CASCADE)
+
+    class Meta:
+        verbose_name = 'Dislike'
+        verbose_name_plural = 'Dislikes'
+
+
+class Followers(models.Model):
```
Comment (revimi) [inline main/models.py:66]: Please add `timestamp` field

File: main/models.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    description = models.TextField(max_length=500, blank=True)
+    published = models.DateTimeField(auto_now_add=True, db_index=True)
+    user = models.ForeignKey('User', on_delete=models.CASCADE)
+
+    class Meta:
+        verbose_name = 'Post'
+        verbose_name_plural = 'Posts'
+
+    def __str__(self):
+        return f'{self.description}'
```
Comment (revimi) [inline main/models.py]: Probably return timestamp as description may be blank or too long

File: main/models.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    nickname = models.CharField(max_length=30, unique=True)
+    personal_information = models.TextField(max_length=300)
+    profile_photo = models.ImageField(null=True, blank=True)
+
+    class Meta:
+        verbose_name = 'User'
+        verbose_name_plural = 'Users'
+
+    def __str__(self):
+        return f'{self.nickname}'
```
Comment (revimi) [inline main/models.py]:
```suggestion
        return self.nickname
```

File: main/models.py [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    class Meta:
+        verbose_name = 'Post'
+        verbose_name_plural = 'Posts'
+
+    def __str__(self):
+        return f'{self.description}'
+
+
+class Comment(models.Model):
+    comment_text = models.TextField(max_length=500)
```
Comment (revimi) [inline main/models.py]:
```suggestion
    text = models.TextField(max_length=500)
```

File: main/models.py [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+    class Meta:
+        verbose_name = 'Post'
+        verbose_name_plural = 'Posts'
+
+    def __str__(self):
+        return f'{self.description}'
+
+
+class Comment(models.Model):
```
Comment (revimi) [inline main/models.py:32]: Please add timestamp field

File: main/models.py [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    comment_text = models.TextField(max_length=500)
+    post = models.ForeignKey('Post', on_delete=models.CASCADE)
+    user = models.ForeignKey('User', on_delete=models.CASCADE)
+
+    class Meta:
+        verbose_name = 'Comment'
+        verbose_name_plural = 'Comments'
+
+    def __str__(self):
+        return f'{self.comment_text}'
```
Comment (revimi) [inline main/models.py]:
```suggestion
        return self.comment_text
```

File: main/models.py [hunk 13]
```diff
@@ -0,0 +1,66 @@
+from django.db import models
```
Comment (revimi) [inline main/models.py:1]: Please generate migrations for models

**Review by revimi — CHANGES_REQUESTED** [review]

File: main/models.py (hunk 2 above)
Comment (marinaRomanchuk) [reply main/models.py]: Done

File: main/models.py (hunk 4 above)
Comment (marinaRomanchuk) [reply main/models.py]: Done

File: main/models.py (hunk 5 above)
Comment (marinaRomanchuk) [reply main/models.py:46]: Done

File: main/models.py (hunk 6 above)
Comment (marinaRomanchuk) [reply main/models.py:56]: Done

File: main/models.py (hunk 7 above)
Comment (marinaRomanchuk) [reply main/models.py:66]: Done

File: main/models.py (hunk 8 above)
Comment (marinaRomanchuk) [reply main/models.py]: Agree, done

File: main/models.py (hunk 9 above)
Comment (marinaRomanchuk) [reply main/models.py]: Done

File: main/models.py (hunk 10 above)
Comment (marinaRomanchuk) [reply main/models.py]: Done

File: main/models.py (hunk 11 above)
Comment (marinaRomanchuk) [reply main/models.py:32]: Done

File: main/models.py (hunk 12 above)
Comment (marinaRomanchuk) [reply main/models.py]: Done

File: main/models.py [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+        verbose_name_plural = 'Users'
+
+    def __str__(self):
+        return self.user.username
+
+
+class Post(models.Model):
+    photo = models.ImageField()
+    description = models.TextField(max_length=500, blank=True)
+    timestamp = models.DateTimeField(auto_now_add=True, db_index=True)
```
Comment (revimi) [inline main/models.py]: Why do we need it to be an index?

File: main/models.py (hunk 3 above)
Comment (marinaRomanchuk) [reply main/models.py]: I made class UserProfile, field user is OneToOneField to Django user model, field 'email' exists in that model, date_of_birth and timestamp have been added to UserProfile

File: main/models.py (hunk 1 above)
Comment (marinaRomanchuk) [reply main/models.py]: Username exists in Django user model, this line was remowed

File: main/models.py (hunk 13 above)
Comment (marinaRomanchuk) [reply main/models.py:1]: Done

File: main/models.py [hunk 15]
```diff
@@ -0,0 +1,71 @@
+from django.db import models
+from django.contrib.auth.models import User
+
+
+class UserProfile(models.Model):
```
Comment (revimi) [inline main/models.py]: Why not inherit it from django's `AbstractUser`?

**Review by revimi — CHANGES_REQUESTED** [review]

File: main/models.py [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    description = models.TextField(max_length=300)
+    date_of_birth = models.DateField()
+    timestamp = models.DateTimeField(auto_now_add=True, db_index=True)
```
Comment (revimi) [inline main/models.py]: `django.contrib.auth.models.User` already has `date_joined` field

File: main/models.py (hunk 14 above)
Comment (marinaRomanchuk) [reply main/models.py]: Actually I don't know, why I thought it can be useful, remowed

File: main/models.py (hunk 15 above)
Comment (revimi) [reply main/models.py]: https://docs.djangoproject.com/en/4.0/topics/auth/customizing/#extending-django-s-default-user

Comment (marinaRomanchuk) [reply main/models.py]: I thought, OneToOneField is possible, fixed

File: main/models.py (hunk 16 above)
Comment (marinaRomanchuk) [reply main/models.py]: Remowed

File: main/migrations/0001_initial.py [hunk 17]
```diff
@@ -0,0 +1,95 @@
+# Generated by Django 3.2.11 on 2022-01-26 13:38
```
Comment (revimi) [inline main/migrations/0001_initial.py]: Please update the migration

File: main/models.py [hunk 18]
```diff
@@ -0,0 +1,69 @@
+from django.db import models
+from django.contrib.auth.models import AbstractUser
+
+
+class UserProfile(AbstractUser):
```
Comment (revimi) [inline main/models.py]:
```suggestion
class User(AbstractUser):
```

**Review by revimi — APPROVED** [review]

File: main/models.py (hunk 18 above)
Comment (marinaRomanchuk) [reply main/models.py]: Done

File: main/migrations/0001_initial.py (hunk 17 above)
Comment (marinaRomanchuk) [reply main/migrations/0001_initial.py]: Done

**Pull request merged**